src/move/econia/sources/market.move

```diff
@@ -3451,11 +3451,10 @@ module econia::market {
             } else if (still_crosses_spread) {
                 option::fill(&mut cancel_reason_option,
                              CANCEL_REASON_MAX_QUOTE_TRADED);
-            } else {
-                if (remaining_size < order_book_ref_mut.min_size) {
-                    option::fill(&mut cancel_reason_option,
-                                 CANCEL_REASON_TOO_SMALL_AFTER_MATCHING);
-                }
+            } else if ((remaining_size > 0) &&
+                       (remaining_size < order_book_ref_mut.min_size)) {
+                option::fill(&mut cancel_reason_option,
+                             CANCEL_REASON_TOO_SMALL_AFTER_MATCHING);
```

Comment:
I think `min_size` should be enforced only when the order is first placed. If an order of a size greater than `min_size` is placed, and it fills in such a way that after the initial fill the remaining size falls below this threshold, that's not an issue.

Reply:
@bogberry addressed as of f2c1777